hadoop-ozone/ozone-recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/FileSizeCountTask.java

```diff
@@ -0,0 +1,231 @@
+        for (FileCountBySize row : resultSet) {
+          upperBoundCount[index] = row.getCount();
+          index++;
+        }
+      }
+    } else {
+      upperBoundCount = new long[maxBinSize];
+    }
+  }
+
+  @Override
+  protected Collection<String> getTaskTables() {
+    return tables;
+  }
+
+  /**
+   * Read the Keys from update events and update the count of files
+   * pertaining to a certain upper bound.
+   *
+   * @param events Update events - PUT/DELETE.
+   * @return Pair
+   */
+  @Override
+  Pair<String, Boolean> process(OMUpdateEventBatch events) {
+    LOG.info("Starting a 'process' run of FileSizeCountTask.");
+    Iterator<OMDBUpdateEvent> eventIterator = events.getIterator();
+
+    fetchUpperBoundCount("process");
+
+    while (eventIterator.hasNext()) {
+      OMDBUpdateEvent<String, OmKeyInfo> omdbUpdateEvent = eventIterator.next();
+      String updatedKey = omdbUpdateEvent.getKey();
+      OmKeyInfo updatedValue = omdbUpdateEvent.getValue();
+
+      try{
+        switch (omdbUpdateEvent.getAction()) {
+        case PUT:
+          updateUpperBoundCount(updatedValue, "PUT");
+          break;
+
+        case DELETE:
+          updateUpperBoundCount(updatedValue, "DELETE");
+          break;
+
+        default: LOG.trace("Skipping DB update event : " + omdbUpdateEvent
+                  .getAction());
+        }
+      } catch (IOException e) {
+        LOG.error("Unexpected exception while updating key data : {} {}",
+                updatedKey, e.getMessage());
+        return new ImmutablePair<>(getTaskName(), false);
+      } finally {
+        populateFileCountBySizeDB();
+      }
+    }
+    LOG.info("Completed a 'process' run of FileSizeCountTask.");
+    return new ImmutablePair<>(getTaskName(), true);
+  }
+
+  /**
+   * Calculate the bin index based on size of the Key.
+   *
+   * @param dataSize Size of the key.
+   * @return int bin index in upperBoundCount
+   */
+  private int calcBinIndex(long dataSize) {
+    if(dataSize >= maxFileSizeUpperBound) {
+      return Integer.MIN_VALUE;
+    } else if (dataSize > SIZE_512_TB) {
+      //given the small difference in 512TB and 512TB + 1B, index for both would
+      //return same, to differentiate specific condition added.
+      return maxBinSize - 1;
+    }
+    int logValue = (int) Math.ceil(Math.log(dataSize)/Math.log(2));
+    if(logValue < 10){
+      return 0;
+    } else{
+      return (dataSize % ONE_KB == 0) ? logValue - 10 + 1: logValue - 10;
+    }
+  }
+
+  private void countFileSize(OmKeyInfo omKeyInfo) throws IOException{
+    int index = calcBinIndex(omKeyInfo.getDataSize());
+    if(index == Integer.MIN_VALUE) {
+      throw new IOException("File Size larger than permissible file size "
+          + maxFileSizeUpperBound +" bytes");
+    }
+    upperBoundCount[index]++;
+  }
+
+  private void populateFileCountBySizeDB() {
+    for (int i = 0; i < upperBoundCount.length; i++) {
+      long fileSizeUpperBound = (long) Math.pow(2, (10 + i));
```

Comment:
Seems we have changed to log2, the computation will be accurate but now what is the plan to show all files < 1 MB query? Seems more computation to be done in UI.

Reply:
Yes @swagle. We can do aggregations in the UI to show whatever ranges the user is interested in. The more data points we have is better for the UI. These aggregates in UI plots are very inexpensive and will improve user experience with richer data points.